reconcile/utils/terrascript_aws_client.py

```diff
@@ -994,7 +994,8 @@ def get_resource_lifecycle(
             if lifecycle.ignore_changes is None:
                 lifecycle.ignore_changes = []
             if "all" in lifecycle.ignore_changes:
-                lifecycle.ignore_changes = "all"
+                # 'all' must be passed as a string! stupid terraform aws provider!
+                lifecycle.ignore_changes = "all"  # type: ignore[assignment]
```

Comment:
`lifecycle.ignore_changes` type is defined from gql, let's not assign wrong types to it, change return value instead:

```python
            ignore_changes = (
                "all" if "all" in lifecycle.ignore_changes else lifecycle.ignore_changes
            )
            return lifecycle.dict(by_alias=True) | {"ignore_changes": ignore_changes}
```